nano/node/block_pipeline/early_verify_filter.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+
+namespace nano
+{
+class block;
+class epochs;
+class stat;
+namespace block_pipeline
+{
+/**
+ *  This class filters blocks by when their signatures can be verified.
+ *  Blocks can be early which means they can be verified without a ledger access, or they can be late which means the ledger must be accessed to determine the signer
+ */
+class early_verify_filter
+{
+public:
+	early_verify_filter (nano::epochs & epochs, nano::stat & stats);
+	void sink (std::shared_ptr<nano::block> block) const;
```

Comment:
Is sink the input of the filter? Do you call sink to provide blocks to the filter?